### PR TITLE
Update Environment.ProcessorCount - port dotnet/runtime#45943

### PR DESCRIPTION
**Pull request by jkotas**
Also delete some dead code

### CODE REVIEW

**Comment by jkotas** [comment]
I have verified that `Environment.ProcessorCount` takes process affinity into account with this change.

File: src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp [hunk 1]
```diff
@@ -126,10 +126,6 @@ static const int tccMilliSecondsToMicroSeconds = 1000;
 static const int tccMilliSecondsToNanoSeconds = 1000000;
 static const int tccMicroSecondsToNanoSeconds = 1000;
 
-static uint32_t g_cNumProcs = 0;
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp:129]: I think we can also delete a couple SYSCONF/SYSCTL related defines and ifdefs at the beginning of this file.

Maybe the can also be deleted from config.h.in and configure.cmake?

File: src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp [hunk 2]
```diff
@@ -35,20 +35,6 @@ uint32_t PalEventWrite(REGHANDLE arg1, const EVENT_DESCRIPTOR * arg2, uint32_t a
 // Index for the fiber local storage of the attached thread pointer
 static uint32_t g_flsIndex = FLS_OUT_OF_INDEXES;
 
-GCSystemInfo g_RhSystemInfo;
-
-bool InitializeSystemInfo()
-{
-    SYSTEM_INFO systemInfo;
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp:42]: Can we delete the SYSTEM_INFO struct from PalRedhawk.h?

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**